lcls-twincat-motion/Library/POUs/Motion/FB_MotionPneumaticActuator.TcPOU

```diff
@@ -0,0 +1,209 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<TcPlcObject Version="1.1.0.1" ProductVersion="3.1.4024.3">
+  <POU Name="FB_MotionPneumaticActuator" Id="{3965ca68-046c-4858-93f1-aaf5b6059f9e}" SpecialFunc="None">
+    <Declaration><![CDATA[(*This function blcok implements a pnuematic actuator. That can be signle or double acting by setting the ibSingleCntrl accordingly*)
+(* with double acting ibCntrlHold signal should be false, while with single acting the signal should be true*)
+FUNCTION_BLOCK FB_MotionPneumaticActuator
+VAR_INPUT
+    (*EPS Interlock Bits*)
+    ibInsertOK: BOOL; (*Actuator can be Inserted*)
+    ibRetractOK: BOOL; (*ACtuator can be retracted*)
+    ibPMPS_OK:BOOL; (*to be linked the Arbiter bit*)
+    ibSingleCntrl:BOOL;(* TRUE if Actuator requires one Output signal to be activated, FALSE if its double acting i.e two outputs are required*)
+    ibCntrlHold:BOOL; (* Control Signal must retain its value, must be TRUE in the case of single acting*)
+    ibOverrideInterlock:BOOL; (*if true interlocks are ignored*)
+    // Reset fault
+    {attribute 'pytmc' := '
+    pv: FF_Reset
+    '}
+    i_xReset: BOOL;
+END_VAR
+VAR_OUTPUT
+    {attribute 'pytmc' := '
+    pv:
+    '}
+    stPeumaticActuator    :    DUT_MotionPneumaticActuator;
+    {attribute 'pytmc' := '
+     pv: MPS_OK
+     field: ZNAM FALSE
+     field: ONAM TRUE
+     field: DESC TRUE if MPS signal is OK
+    '}
+    xMPS_OK:BOOL;
+END_VAR
+VAR_IN_OUT
+    io_fbFFHWO    :    FB_HardwareFFOutput;
+END_VAR
+VAR
+    // PMPS
+    fbFF    :    FB_FastFault :=(
+        i_DevName := 'MPA',
+        i_Desc := 'Fault occurs when the device is moving',
+        i_TypeCode := 10#1010);
+
+    (*Init*)
+    xFirstPass    :    BOOL;
+    fbFSInit        :    R_TRIG;
+
+    (* Timeouts*)
+    tTimeOutDuration: TIME:= T#10S;
+    tInserttimeout: TON;
+    tRetracttimeout:TON;
+    (*Limit switch latch timer*)
+    tLimitSwitchLatchDuration: TIME:=T#1S;
+    tInsertLimitSwitch:TON;
+    tRetractLimitSwitch:TON;
+
+    (*Logging*)
+    fbLogger : FB_LogMessage := (eSubsystem:=E_SubSystem.MOTION);
+    ePrevState : ENUM_PnuematicActuatorPositionState;
+    tAction : R_TRIG; // Primary action of this device (Insert_DO, Retract_DO, etc.)
+    tOverrideActivated : R_TRIG;
+
+    (*IO*)
+    i_xInsertedLS    AT%I*: BOOL;
+    i_xRetractedLS    AT%I*: BOOL;
+    q_xInsert_DO    AT%Q*: BOOL;
+    q_xRetract_DO    AT%Q*: BOOL;
```

Comment:
I see now, the links are down here on the function block, not up there on the DUT